pilot/pkg/controller/workloadentry/workloadentry_controller.go

```diff
@@ -675,3 +763,46 @@ func (c *Controller) handleErr(err error, key interface{}) {
 func makeProxyKey(proxy *model.Proxy) string {
 	return string(proxy.Metadata.Network) + proxy.IPAddresses[0]
 }
+
+func lookupWorkloadEntryByProxyID(store model.ConfigStoreCache, proxy *model.Proxy) string {
+	if len(proxy.IPAddresses) == 0 {
+		return ""
+	}
+
+	parts := strings.Split(proxy.ID, ".")
+	if len(parts) != 2 || parts[1] != proxy.Metadata.Namespace {
+		return ""
+	}
+	name := parts[0]
```

Comment:
So this also adds a restriction that name of WE must match, right?

Reply:
For the purposes of this feature, I need logic that finds `WorkloadEntry` corresponding to the connected proxy.

Right now this code, indeed, has only 1 strategy to find matching `WorkloadEntry`.

Ideally, I would like to reuse logic of [kube/controller](https://github.com/istio/istio/blob/06f42d2b95f14c00e6a989370c7975c552da8643/pilot/pkg/serviceregistry/kube/controller/controller.go#L1057-L1060), but I didn't find a way to achieve that.

Is it possible to introduce dependency on `kube/controller` in here?

Reply:
@howardjohn BTW, I've opened [another PR](https://github.com/istio/istio/pull/37129) to enhance how `kube/controller` searches for a matching `WorkloadEntry`.

Reply:
this function has been removed